src/DB/Migration/Migration20211228T1640692399.php

```diff
@@ -49,11 +58,24 @@ public function get_applicable_version(): string {
 	 * @return void
 	 */
 	public function apply(): void {
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 		if ( $this->shipping_rate_table->exists() ) {
-			// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
-			// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 			$this->wpdb->query( "ALTER TABLE `{$this->wpdb->_escape( $this->shipping_rate_table->get_name() )}` ALTER COLUMN `method` DROP DEFAULT" );
-			// phpcs:enable WordPress.DB.PreparedSQL.NotPrepared
+
+			$mc_settings = $this->options->get( Options::MERCHANT_CENTER );
+			if ( isset( $mc_settings['offers_free_shipping'] ) && false !== boolval( $mc_settings['offers_free_shipping'] ) && isset( $mc_settings['free_shipping_threshold'] ) ) {
+				// Move the free shipping threshold from the options to the shipping rate table.
+				$serialized_options = json_encode( [ 'free_shipping_threshold' => (float) $mc_settings['free_shipping_threshold'] ] );
+				$this->wpdb->query( $this->wpdb->prepare( "UPDATE `{$this->wpdb->_escape( $this->shipping_rate_table->get_name() )}` SET `options`=%s WHERE `method` = 'flat_rate'", $serialized_options ) );
+
+				// Remove the free shipping threshold from the options.
+				unset( $mc_settings['free_shipping_threshold'] );
+				unset( $mc_settings['offers_free_shipping'] );
+				$this->options->update( Options::MERCHANT_CENTER, $mc_settings );
```

Comment:
These are only unset if we had free shipping setup. We always saved the values regardless of whether they were true or not, so can we unset these outside the if condition?

Reply:
Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1305/commits/54aef1114917f6ec36a821742d84ab87b7b2c599